File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
        Map<String, Integer> map = new HashMap<>();
        
        for (String key: keymap){
            for (int k = 0; k < key.length(); k++){
                String m = key.substring(k, k+1);
                if (map.containsKey(m)){
                    if (map.get(m) > k+1) map.put(m, k+1);
                } else {
                    map.put(m, k+1);
                }
            }            
        }
        
        for (int i = 0; i< targets.length; i++){
            String target = targets[i];
            int tmp = 0;
            for (String t: target.split("")){
                if (map.containsKey(t)){
                    tmp += map.get(t);
                } else {
                    tmp = -1;
                    break;
                }
            }
            answer[i] = tmp; 
        }
        return answer;
    }
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    d = {}
    
    for key in keymap:
        for k in range(len(key)):
            if d.get(key[k]):
                if d[key[k]] > k+1:
                    d[key[k]] = k+1
            else:
                d[key[k]] = k+1
                    
    for target in targets:
        tmp = 0
        for t in target:
            if d.get(t):
                tmp += d[t]
            else:
                tmp = -1
                break
        answer.append(tmp)
        
    return answer